### PR TITLE
Improve VM control by owner

### PR DESCRIPTION
**Pull request by hoh**
- WIP: Fix control enpoints
- WIP: Use pydantic classes in control endpoints
- WIP: Use pydantic classes in control endpoints

### CODE REVIEW

**Comment by BjrInt** [comment]
# Todo

- [ ] Verify that the SignedOperation `time` field is within an "acceptable range"
- [ ] Verify that the SignedOperation `path` field matches the route

**Review by BjrInt — APPROVED** [review]

File: src/aleph/vm/orchestrator/views/operator.py [hunk 1]
```diff
@@ -165,19 +102,24 @@ async def stream_logs(request: web.Request):
...
+    except (KeyError, ValueError) as error:
+        raise web.HTTPBadRequest(body="Invalid timeout duration") from error
     if not 0 < timeout < timedelta(days=10).total_seconds():
         return web.HTTPBadRequest(body="Invalid timeout duration")
 
     pool: VmPool = request.app["vm_pool"]
     execution = get_execution_or_404(vm_hash, pool=pool)
 
+    if not is_sender_authorized(authenticated_sender, execution.message):
+        return web.Response(status=401, body="Unauthorized sender")
```
Comment (MHHukiewitz) [inline src/aleph/vm/orchestrator/views/operator.py:121]: Same as above, should be 403

File: src/aleph/vm/orchestrator/views/operator.py [hunk 2]
```diff
@@ -195,6 +137,12 @@ async def operate_stop(request: web.Request):
     logger.debug(f"Iterating through running executions... {pool.executions}")
     execution = get_execution_or_404(vm_hash, pool=pool)
 
+    if not is_sender_authorized(authenticated_sender, execution.message):
+        return web.Response(status=401, body="Unauthorized sender")
+
+    if not is_sender_authorized(authenticated_sender, execution.message):
+        return web.Response(status=401, body="Unauthorized sender")
```
Comment (MHHukiewitz) [inline src/aleph/vm/orchestrator/views/operator.py:144]: HTTP 403

File: src/aleph/vm/orchestrator/views/operator.py [hunk 3]
```diff
@@ -205,28 +153,39 @@ async def operate_stop(request: web.Request):
...
     """
     vm_hash = get_itemhash_or_400(request.match_info)
     pool: VmPool = request.app["vm_pool"]
     execution = get_execution_or_404(vm_hash, pool=pool)
 
-    # TODO: implement this endpoint
-    logger.info(f"Rebooting {execution.vm_hash}")
-    return web.Response(status=200, body=f"Rebooted {execution.vm_hash}")
+    if not is_sender_authorized(authenticated_sender, execution.message):
+        return web.Response(status=401, body="Unauthorized sender")
```
Comment (MHHukiewitz) [inline src/aleph/vm/orchestrator/views/operator.py:165]: 403

File: src/aleph/vm/orchestrator/views/operator.py [hunk 4]
```diff
@@ -205,28 +153,39 @@ async def operate_stop(request: web.Request):
...
+async def operate_erase(request: web.Request, authenticated_sender: str) -> web.Response:
     """Delete all data stored by a virtual machine.
     Stop the virtual machine first if needed.
     """
     vm_hash = get_itemhash_or_400(request.match_info)
     pool: VmPool = request.app["vm_pool"]
     execution = get_execution_or_404(vm_hash, pool=pool)
 
+    if not is_sender_authorized(authenticated_sender, execution.message):
+        return web.Response(status=401, body="Unauthorized sender")
```
Comment (MHHukiewitz) [inline src/aleph/vm/orchestrator/views/operator.py:187]: 403

**Review by MHHukiewitz — COMMENTED** [review]
Pretty good, just a few comments on the returned status codes

**Pull request merged**